Assets/HoloToolkit/Utilities/Scripts/HeadsUpDirectionIndicator.cs

```diff
@@ -0,0 +1,313 @@
+                }
+                else
+                {
+                    return FrustumPlanes.Right;
+                }
+            }
+        }
+ 
+        // given a frustum wall we wish to snap the pointer to, this function returns a ray
+        // along which the pointer should be placed to appear at the appropiate point along
+        // the edge of the indicator field.
+        private bool TryGetIndicatorPosition(Vector3 targetPosition, Camera camera, Plane frustumWall, out Ray r)
+        {
+            // Think of the pointer as pointing the shortest rotation a user must make to see a
+            // target. The shortest rotation can be obtained by finding the great circle defined
+            // be the target, the camera position and the center position of the view. The tangent
+            // vector of the great circle points the direction of the shortest rotation. This
+            // great circle and thus any of it's tangent vectors are coplanar with the plane
+            // defined by these same three points.
+            Vector3 cameraToTarget = targetPosition - camera.transform.position;
+            Vector3 normal = Vector3.Cross(cameraToTarget.normalized, camera.transform.forward);
+
+            // In the case that the three points are colinear we cannot form a plane but we'll
+            // assume the target is directly behind us and we'll use a prechosen plane.
+            if (normal == Vector3.zero)
+            {
+                normal = -Vector3.right;
+            }
+
+            Plane q = new Plane(normal, targetPosition);
+            return TryIntersectPlanes(frustumWall, q, out r);
+        }
+
+        // Obtain the line of intersection of two planes. This is based on a method
+        // described in the GPU Gems series.
+        private bool TryIntersectPlanes(Plane p, Plane q, out Ray intersection)
+        {
+            Vector3 rNormal = Vector3.Cross(p.normal, q.normal);
+            float det = rNormal.sqrMagnitude;
+
+            if (det != 0.0f)
+            {
+                Vector3 rPoint = ((Vector3.Cross(rNormal, q.normal) * p.distance) +
+                    (Vector3.Cross(p.normal, rNormal) * q.distance)) / det;
+                intersection = new Ray(rPoint, rNormal);
+                return true;
+            }
+            else
+            {
+                intersection = new Ray();
+                return false;
+            }
+        }
+
+        // Modify the pointer location and orientation to point along the shortest rotation,
+        // toward tergetPosition, keeping the pointer confined inside the frustum defined by
+        // planes.
+        private void UpdatePointerTransform(Camera camera, Plane []planes, Vector3 targetPosition)
+        {
+            // Start by assuming the pointer should be placed at the target position.
+            Vector3 indicatorPosition = camera.transform.position + Depth * (targetPosition - camera.transform.position).normalized;
```

Comment:
cache the camera position rather than calling camera.trasform.position each time through the function.  Profiling shows that calls to get the camera transform are adding up.

Function Name	Total CPU (%)	Self CPU (%)	Total CPU (ms)	Self CPU (ms)	Module
 - HoloToolKit.Unity.HeadsUpDirectionIndicator::UpdatePointerTransform	1.62 %	0.00 %	5	0	assembly-csharp.dll
   - UnityEngineProxy.InternalCalls::Component_Get_Custom_PropTransform	0.65 %	0.00 %	2	0	unityengineproxy.dll
     - Component_Get_Custom_PropTransform	0.65 %	0.00 %	2	0	unityenginedelegates.dll
         Component_Get_Custom_PropTransform	0.65 %	0.65 %	2	2	unityplayer.dll